lib/model/unittest/CSearchKeyTest.cc

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include "CSearchKeyTest.h"
+
+#include <model/CSearchKey.h>
+
+using namespace ml;
+using namespace model;
+
+void CSearchKeyTest::testSimpleCountComesFirst() {
+    CSearchKey key1{1};
+    CSearchKey key2{2};
+    CPPUNIT_ASSERT(CSearchKey::simpleCountKey() < key1);
+    CPPUNIT_ASSERT(CSearchKey::simpleCountKey() < key2);
```

Comment:
nit: this seems a bit on the weak side. I mean the sort implementation change is clearly correct, but maybe randomly choose from multiple different input parameter values if you're going to unit test, i.e. different `function_t::EFunction`, `useNull`, `model_t::EExcludeFrequent` and field name strings. `test::CRandomNumbers::generateWords` could be used for the last part and the other.